src/middlewared/middlewared/plugins/reporting/events.py

```diff
@@ -366,6 +333,61 @@ def _cpu_temperature(self):
 
         return temperature
 
+    AMD_CORE_COUNT = None
+    AMD_CPU_MODEL = None
+    AMD_PREFER_TDIE = (
+        # https://github.com/torvalds/linux/blob/master/drivers/hwmon/k10temp.c#L121
+        # static const struct tctl_offset tctl_offset_table[] = {
+        "AMD Ryzen 5 1600X",
+        "AMD Ryzen 7 1700X",
+        "AMD Ryzen 7 1800X",
+        "AMD Ryzen 7 2700X",
+        "AMD Ryzen Threadripper 19",
+        "AMD Ryzen Threadripper 29",
+    )
+
+    def _amd_cpu_temperature(self, amd_sensor):
+        if self.AMD_CPU_MODEL is None:
+            self.AMD_CPU_MODEL = self.middleware.call_sync('system.info')['model']
+
+        if self.AMD_CORE_COUNT is None:
+            self.AMD_CORE_COUNT = len([
```

Comment:
We already cache the CPU core count and the CPU model in `system.cpu_info` so I'd rather use that instead since it removes a subprocess call.